### ISSUE TITLE
Enhance a11y for `Pagination`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#5396]
name: Pagination

### Description
4 accessibility violation was found in the `Pagination`.

#### [Components / Pagination / Variant](https://yamada-ui.github.io/yamada-ui/?path=/story/components-pagination--variant)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Pagination / Size](https://yamada-ui.github.io/yamada-ui/?path=/story/components-pagination--size)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Pagination / Text](https://yamada-ui.github.io/yamada-ui/?path=/story/components-pagination--text)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

#### [Components / Pagination / CustomIcon](https://yamada-ui.github.io/yamada-ui/?path=/story/components-pagination--custom-icon)

Received:
Landmarks should have a unique role or role/label/title (i.e. accessible name) combination (landmark-unique)

Fix any of the following:
  The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.11/landmark-unique?application=axeAPI

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.
